### PR TITLE
update docker webapp guide for carbon

### PR DESCRIPTION
**Pull request by jbottigliero**
- updates refrences of boron for carbon
- updates express package version in example (4.16.1 is latest stable)
- updates Dockerfile `COPY` (and instructions) to assume `npm@5+` and warn about `npm@4-`
- updates various stdout examples

### CODE REVIEW

File: locale/en/docs/guides/nodejs-docker-webapp.md [hunk 1]
```diff
@@ -140,15 +145,15 @@ CMD [ "npm", "start" ]
...
+FROM node:carbon
 
 # Create app directory
 WORKDIR /usr/src/app
 
 # Install app dependencies
-COPY package.json .
-# For npm@5 or later, copy package-lock.json as well
-# COPY package.json package-lock.json ./
+COPY package.json package-lock.json ./
```
Comment (jbottigliero) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: I considered making this a wildcard to account for both cases... but thought this was more straight-forward and consistent what was here before.

The following will _not_ fail if the `package-lock.json` file does not exist (using an older version of `npm` or if `npm install` was not run before `docker build ...`:
```
COPY package.json package-*.json ./
```

**Comment by jbottigliero** [comment]
If this looks good, I can at the very least update the examples/stdout in the translation files if that is desired.

File: locale/en/docs/guides/nodejs-docker-webapp.md (hunk 1 above)
Comment (fhemberger) [reply locale/en/docs/guides/nodejs-docker-webapp.md]: At this point, `COPY .` should be enough IMHO. No need to explicitly copy `package*.json` again.

**Review by fhemberger — CHANGES_REQUESTED** [review]

**Comment by fhemberger** [comment]
Otherwise LGTM.

File: locale/en/docs/guides/nodejs-docker-webapp.md (hunk 1 above)
Comment (jbottigliero) [reply locale/en/docs/guides/nodejs-docker-webapp.md]: Just to clarify, are you suggesting to forego the explicit coping of `package*.json` files and simply `COPY . .` _before_ the `npm install` (removing the `# Bundle app source` command)? Resulting in:
```
FROM node:carbon

# Create app directory
WORKDIR /usr/src/app

# Bundle app source
COPY . .

# Install app dependencies
RUN npm install
# If you are building your code for production
# RUN npm install --only=production

EXPOSE 8080
CMD [ "npm", "start" ]
```

Or, replacing the replacing the `COPY package.json package-lock.json ./` with `COPY package*.json .`?

Comment (fhemberger) [reply locale/en/docs/guides/nodejs-docker-webapp.md]: Sorry, my mistake, got a bit mixed up in the diff. I meant 

> replacing the replacing the `COPY package.json package-lock.json ./` with `COPY package*.json .`?

**Comment by jbottigliero** [comment]
Updated to use `COPY pacakge*.json .`.

For the additional locale versions, is it worth updating the `boron` references and docker commands leaving the current comments/verbiage intact? 

> note that if you are using `npm` version 5 or later you will also want to copy
 `package-lock.json`, which is generated once you run `npm install`:

The above seems like it would be the only misleading copy from the original (translated) version.

**Comment by fhemberger** [comment]
@jbottigliero Sounds good to me. :+1:

**Review by fhemberger — APPROVED** [review]

**Review by lpinca — APPROVED** [review]

**Review by hiroppy — APPROVED** [review]

**Comment by jbottigliero** [comment]
I've opened #1476 adding the updates [mentioned in this comment](https://github.com/nodejs/nodejs.org/pull/1467#issuecomment-344810234) along a few URL updates.

**Pull request merged**